test/import/import-cli-config-sync.test.ts

```diff
@@ -0,0 +1,95 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as yaml from 'yaml';
+import { readConfigSync, addImportToConfig } from '../../src/config';
+import { Language } from '../../src/import/base';
+import { ManifestObjectDefinition } from '../../src/import/crd';
+import { importDispatch } from '../../src/import/dispatch';
+
+const configFilePath = path.join(__dirname, 'configOutput');
+const jenkinsCRD = 'https://raw.githubusercontent.com/jenkinsci/kubernetes-operator/master/deploy/crds/jenkins.io_jenkins_crd.yaml';
+
+async function withTempFixture(data: any, test: (fixture: string, configFile: string, cwd: string) => Promise<void>) {
+  const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdk8s-import-test'));
+  const fixture = path.join(tempDir, 'fixture.yaml');
+  try {
+    if (Array.isArray(data)) {
+      fs.writeFileSync(fixture, data.map(d => yaml.stringify(d)).join('\n---\n'));
+    } else {
+      fs.writeFileSync(fixture, yaml.stringify(data));
+    }
+    await test(fixture, configFilePath, tempDir);
+  } finally {
+    fs.removeSync(tempDir);
+    // clean up cdk8s.yaml file, reseting back to template:
+    const defaultConfig = yaml.parse(fs.readFileSync(path.join(configFilePath, 'cdk8s-template.yaml'), 'utf-8'));
+    await fs.outputFile(path.join(configFilePath, 'cdk8s.yaml'), yaml.stringify(defaultConfig));
+  }
+}
+
+test('reading default config file works', async () => {
+
+  // clean up cdk8s.yaml file, reseting back to template:
+  const defaultConfig = yaml.parse(fs.readFileSync(path.join(configFilePath, 'cdk8s-template.yaml'), 'utf-8'));
+  await fs.outputFile(path.join(configFilePath, 'cdk8s.yaml'), yaml.stringify(defaultConfig));
+  const config1 = readConfigSync(configFilePath);
+  expect(config1.language).toEqual('typescript');
+  expect(config1.app).toEqual('node main.js');
+  expect(config1.imports?.length == 1).toBeTruthy();
+  expect(config1.imports?.includes('k8s')).toBeTruthy();
+});
+
+test('writing default config file works', async () => {
+  await addImportToConfig(jenkinsCRD, configFilePath);
```

Comment:
This isn't the right thing to invoke in the test. What if we by mistake remove the call to `addImportToConfig` from the `importDispatch` function? This test would pass, but the functionality will be broken. 

Ideally, the test should invoke `cdk8s import` - just like a user would do. Since its a little awkward, we can invoke `importDispatch` and test its result. This is sufficient because if we remove `importDispatch` from the `cdk8s import` command by mistake, our init tests will fail.